app/models/lookup_key.rb

```diff
@@ -158,6 +158,11 @@ def path_elements
     end
   end
 
+  def overriden_on_host?(host)
+    return false if host.is_a?(Hostgroup)
+    lookup_values.find_by_match("fqdn=#{host.fqdn}").present?
```

Comment:
cant you simply do where lookup_values.detect {|lv| lv.lookup_key_id == id} ? (or similar)